quantum-viz/quantum_viz/qiskit_parser.py

```diff
@@ -0,0 +1,370 @@
+import json
+from enum import IntEnum
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+try:
+    import qiskit  # noqa: F401
+except ImportError:
+    raise ImportError(
+        '`qiskit` was not found, try to `pip install "quantum-viz[qiskit]"`'
+    )
+
+from qiskit.circuit import QuantumCircuit, Qubit, Clbit, ClassicalRegister
+from qiskit.circuit.instruction import Instruction
+from qiskit.circuit.controlledgate import ControlledGate
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.barrier import Barrier
+from qiskit.circuit.reset import Reset
+from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.library import (
+    IGate,
+    XGate,
+    YGate,
+    ZGate,
+    HGate,
+    UGate,
+    U1Gate,
+    U2Gate,
+    U3Gate,
+    PhaseGate,
+    RGate,
+    SwapGate,
+    RXGate,
+    RYGate,
+    RZGate,
+    RXXGate,
+    RYYGate,
+    RZXGate,
+    RZZGate,
+    iSwapGate,
+    DCXGate,
+    SGate,
+    SdgGate,
+    TGate,
+    TdgGate,
+    SXGate,
+    SXdgGate,
+)
+from qiskit.circuit.quantumcircuitdata import QuantumCircuitData
+
+INSTRUCTION_TYPE_TO_NAME: Dict[Type[Instruction], str] = {
+    IGate: "I",
+    XGate: "X",
+    YGate: "Y",
+    ZGate: "Z",
+    HGate: "H",
+    UGate: "U",
+    U1Gate: "U1",
+    U2Gate: "U2",
+    U3Gate: "U3",
+    PhaseGate: "P",
+    RGate: "R",
+    SwapGate: "SWAP",
+    RXGate: "Rx",
+    RYGate: "Ry",
+    RZGate: "Rz",
+    RXXGate: "Rxx",
+    RYYGate: "Ryy",
+    RZXGate: "Rzx",
+    RZZGate: "Rzz",
+    iSwapGate: "Iswap",
+    DCXGate: "Dcx",
+    SGate: "S",
+    SdgGate: "Sdg",
+    TGate: "T",
+    TdgGate: "Tdg",
+    SXGate: "√X",
+    SXdgGate: "√Xdg",
+    Measure: "measure",
+}
+
+X_GATE_NAME = INSTRUCTION_TYPE_TO_NAME[XGate]
+MEASURE_NAME = INSTRUCTION_TYPE_TO_NAME[Measure]
+
+
+class RegisterType(IntEnum):
+    QUBIT = 0
+    CLASSICAL = 1
+
+
+class ConditionalRender(IntEnum):
+    ALWAYS = 0
+    ON_ZERO = 1
+    ON_ONE = 2
+    AS_GROUP = 3
+
+
+def qiskit2json(circ: QuantumCircuit, indent=2, **kwargs) -> str:
```

Comment:
I'm not sure we need this method. I think I know why you added (the `Viewer` has a bug that it's not converting the Python dictionary into json, which makes display to fail sometimes), but I'd rather us fixing that.

Reply:
Deleted in https://github.com/jond01/quantum-viz.js/pull/7